remoulade/state/backends/postgres.py

```diff
@@ -0,0 +1,114 @@
+import datetime
+from typing import Any, List
+
+from pytz import utc
+from sqlalchemy import Column, DateTime, SmallInteger, String, asc, create_engine, desc, inspect
+from sqlalchemy.dialects.postgresql import JSONB
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.orm import sessionmaker
+
+from remoulade.state import State, StateBackend
+
+Base: Any = declarative_base()
+
+
+class StoredState(Base):
+    def __init__(self, state: State):
+        for key, value in state.as_dict().items():
+            setattr(self, key, value)
+
+    __tablename__ = "states"
+
+    message_id = Column(String, primary_key=True)
+    status = Column(String)
+    actor_name = Column(String)
+    args = Column(JSONB)
+    kwargs = Column(JSONB)
+    options = Column(JSONB)
+    priority = Column(SmallInteger)
+    enqueued_datetime = Column(DateTime)
+    started_datetime = Column(DateTime)
+    end_datetime = Column(DateTime)
+    group_id = Column(String)
+
+    def as_state(self) -> State:
+        state_dict = {}
+        mapper = inspect(StoredState)
+        for column in mapper.attrs:
+            if getattr(self, column.key) is not None:
+                if column.key in ["enqueued_datetime", "started_datetime", "end_datetime"]:
+                    state_dict[column.key] = utc.localize(getattr(self, column.key)).isoformat()
+                else:
+                    state_dict[column.key] = getattr(self, column.key)
+        return State.from_dict(state_dict)
+
+
+class PostgresBackend(StateBackend):
+    def __init__(self, *, namespace="remoulade-state", encoder=None, client=None, url=None):
+        super().__init__(namespace=namespace, encoder=encoder)
+        db_string = "postgresql://wiremind_owner@localhost:5432/wiremind"
+        db = create_engine(db_string)
+        sm = sessionmaker(db)
+        self.session = sm()
+
+    def get_state(self, message_id):
+        state = self.session.query(StoredState).filter_by(message_id=message_id).first()
+        if not state:
+            return None
+        return state.as_state()
+
+    def set_state(self, state, ttl=3600):
+        stored_state = self.session.query(StoredState).filter_by(message_id=state.message_id).first()
+        if stored_state is None:
+            stored_state = StoredState(state)
+            self.session.add(stored_state)
+        else:
+            for key, value in state.as_dict().items():
+                setattr(stored_state, key, value)
+
+        self.session.commit()
+
+    def get_states(
+        self,
+        *,
+        size: int = 50,
+        offset: int = 0,
+        selected_actors: List[str] = None,
+        selected_statuses: List[str] = None,
+        selected_ids: List[str] = None,
+        start_datetime: datetime.datetime = None,
+        end_datetime: datetime.datetime = None,
+        sort_column: str = None,
+        sort_direction: str = None,
+        **kwargs,
```

Comment:
not sure it is a good idea to keep kwargs here, it can hide errors. Maybe we should use the same API for `get_states` everywhere